### PR TITLE
Aruba central collector

### PR DESCRIPTION
**Pull request by itssapir**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-10441](https://jira-dc.paloaltonetworks.com/browse/CIAC-10441)

## Description
Event collector for Aruba Central.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+        try:
+            token_resp = self._http_request(
+                method='POST',
+                url_suffix='/oauth2/token',
+                headers=headers,
+                params=params,
+            )
+
+        except DemistoException as e:
+            if "Invalid refresh_token" in e.message:
```
Comment (bziser) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py:103]:
```suggestion
            if "Invalid refresh_token" in str(e):
```
Using str(e) ensures that the code works correctly in Python 3 and avoids potential issues with missing attributes.

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+
+        try:
+            response = self._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                params=params,
+                headers=headers,
+            )
+        except DemistoException as e:
+            if 'access token is invalid' in e.message:
```
Comment (bziser) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py:247]:
```suggestion
            if 'access token is invalid' in str(e):
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,724 @@
...
+
+    return next_run
+
+
+def add_time_to_events(events: list[dict] | None, time_arg: str):
+    """
+    Adds the _time key to the events.
+
+    Args:
+        events: List[Dict] - list of events to add the _time key to.
```
Comment (bziser) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py:467]:
```suggestion
        events: List[Dict] - list of events to add the _time key to.
        time_arg: str - the key to be used for time extraction.
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py [hunk 4]
```diff
@@ -0,0 +1,724 @@
...
+            url_suffix='/oauth2/authorize/central/api/login',
+            headers=headers,
+            params=params,
+            json_data=json_data,
+            resp_type='response',
+        )
+        csrf_token = response.cookies.get('csrftoken')
+        session = response.cookies.get('session')
+        if not csrf_token or not session:
+            raise DemistoException('Failed to acquire CSRF token and session from login request, check the credentials are valid')
```
Comment (bziser) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py:154]:
```suggestion
            raise DemistoException('Failed to acquire CSRF token and session from login request. Check if the credentials are valid.')
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+configuration:
+- display: Server URL
+  name: url
+  required: true
+  type: 0
+  section: Connect
+- name: credentials
+  display: Client ID
+  required: true
+  defaultvalue:
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml:17]: If there is no default value, should this line be removed?

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  type: 9
+  displaypassword: Client Secret
+  section: Connect
+  hiddenusername: false
+- name: customer_id
+  required: true
+  defaultvalue:
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml:24]: If there is no default value, should this line be removed?

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  type: 9
+  displaypassword: Customer ID
+  section: Connect
+  hiddenusername: true
+- name: user
+  display: Username
+  required: true
+  defaultvalue:
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.yml:32]: If there is no default value, should this line be removed?

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+## HPE Aruba Central Help
+
+### How to generate Client ID and Secret
+
+In order for the collector to access the Aruba Central API, it must first be added as an application in the Aruba Central API gateway. Doing so will generate a unique pair of client id and secret to be used for authentication. Here are the steps to do so:
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md:5]:
```suggestion
In order for the collector to access the Aruba Central API, it must first be added as an application in the Aruba Central API gateway. Doing so will generate a unique pair of client ID and secret to be used for authentication. 
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+1. Go to the Aruba Central portal and navigate to **Accounts Home** -> **Global Settings** -> **API Gateway**.
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md:7]:
```suggestion
1. Go to the Aruba Central portal and navigate to **Accounts Home** > **Global Settings** > **API Gateway**.
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+- Admin users:
+  - Navigate to **System Apps & Tokens**.
+- Non-admin users:
+  - Navigate to **My Apps & Tokens**.
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md:12]:
```suggestion
   - Admin users: Navigate to **System Apps & Tokens**.
   - Non-admin users: Navigate to **My Apps & Tokens**.
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+2. Click **+ Add Apps & Tokens**
+3. Fill in the required details and click **Generate**
+4. Once created, the new credentials can be viewed in the **My Apps & Tokens** tab
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector_description.md:16]:
```suggestion
2. Click **+ Add Apps & Tokens**.
3. Fill in the required details and click **Generate**.
4. Once created, the new credentials can be viewed in the **My Apps & Tokens** tab.
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/README.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | Server URL | Domain URL for API gateway access | True |
+    | Client ID || True |
+    | Client Secret || True |
+    | Customer ID || True |
+    | Username || True |
+    | Password || True |
+    | Fetch networking events | Whether to fetch networking events or only audit | False |
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/README.md:17]:
```suggestion
    | Fetch networking events | Whether to fetch networking events or only audit logs. | False |
```

File: Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/README.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+`aruba-central-get-events`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| should_push_events | If true, the command will create events, otherwise it will only display them. Possible values are: true, false. Default is false. | Required | 
+| limit | Maximum number of results to return. | Required | 
+| from_date | Date from which to get events. Default is 3 hours prior | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/README.md:45]:
```suggestion
| from_date | Date from which to get events. Default is 3 hours prior. | Optional | 
```

File: Packs/HPEArubaCentral/README.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
+# HPE Aruba Central
+ 
+Aruba Central provides a centralized platform for the management and monitoring of network infrastructure.
+It includes comprehensive event collection and audit log management capabilities, allowing to maintain detailed records of network changes, user activities, and security events.
```
Comment (ShirleyDenkberg) [inline Packs/HPEArubaCentral/README.md:4]:
```suggestion
It includes comprehensive event collection and audit log management capabilities, allowing you to maintain detailed records of network changes, user activities, and security events.
```

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Review by bziser — APPROVED** [review]

**Pull request merged**